src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -632,14 +632,13 @@ void CascadiaSettings::_LayerOrCreateProfile(const Json::Value& profileJson)
     auto profileIndex{ _FindMatchingProfileIndex(profileJson) };
     if (profileIndex)
     {
-        // add a new inheritance layer, and apply json values to child
         auto parentProj{ _profiles.GetAt(*profileIndex) };
         auto parent{ winrt::get_self<Profile>(parentProj) };
-        auto childImpl{ parent->CreateChild() };
-        childImpl->LayerJson(profileJson);
 
-        // replace parent in _profiles with child
-        _profiles.SetAt(*profileIndex, *childImpl);
+        // We don't actually need to CreateChild() here.
+        // When we loaded Profile.Defaults, we created an empty child already.
+        // So this just populates the empty child
+        parent->LayerJson(profileJson);

```

Comment:
this is only guaranteed _if we had a `p.d.`!!!_ otherwise it will layer into the existing profiles!